### PR TITLE
Migrate `server` module tests to JUnit 5, part 4

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR continues migration of the `server` module to the new JUnit 5 test structure.

The following packages were migrated:
1. `entity`

### CODE REVIEW

File: server/src/test/java/io/spine/server/entity/EntityTest.java [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author Alexander Litus
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/EntityTest.java]: Please append yourself as an author.

File: server/src/test/java/io/spine/server/entity/EntityTest.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+            entityNew.setDeleted(true);
+
+            // This should pass.
+            entityNew.checkNotArchived();
+
+            assertThrows(CannotModifyDeletedEntity.class, () -> entityNew.checkNotDeleted());
+        }
+    }
+}
+
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/EntityTest.java]: This line is redundant.

File: server/src/test/java/io/spine/server/entity/TransactionalEntityTest.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+        final Transaction tx = spy(mock(Transaction.class));
+        when(tx.isActive()).thenReturn(true);
+        when(tx.isStateChanged()).thenReturn(txChanged);
+        when(tx.getEntity()).thenReturn(entity);
+
+        entity.injectTransaction(tx);
+        return entity;
+    }
+
+    private static class TeEntity
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/TransactionalEntityTest.java]: It should be somewhere in the `...TestEnv`.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: server/src/test/java/io/spine/server/entity/storage/given/ColumnRecordsTestEnv.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ColumnRecordsTestEnv {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/entity/storage/given/ColumnRecordsTestEnv.java]: Please review the PR and append missing `@author` tag values. In most cases that's going to be you :)

**Pull request merged**